crnlib/lzma/Bcj2Enc.c

```diff
@@ -486,7 +486,7 @@ void Bcj2Enc_Encode(CBcj2Enc *p)
       // (0 < p->tempPos <= 4)
       do
         p->temp[i] = src[i];
-      while (++i != rem);
+      while (++i != rem + 1);

```

Comment:
This doesn't any make any sense. How would you fix a buffer overflow by making it copy MORE data?

I suggest not making any changes to 3rd-party libraries unless it is truly necessary.